### PR TITLE
Feature/gb info add doxygen for gb_info.f and g2grids.f

### PR DESCRIPTION
**Pull request by Hang-Lei-NOAA**
Part of #55.

### CODE REVIEW

File: src/g2grids.f [hunk 1]
```diff
@@ -1,46 +1,33 @@
+!>    @file
+!>    @brief This Fortran Module allows access to predefined GRIB2 Grid
+!>    Definition Templates stored in a file.
+!>    @author Gilbert ORG: W/NP11 @date 2004-04-27
```
Comment (edwardhartnett) [inline src/g2grids.f:4]: Change the author tags to include Gilbert's first name, Stephen. So:
`    @author Stephen Gilbert ORG: W/NP11 @date 2004-04-27
`

File: src/g2grids.f [hunk 2]
```diff
@@ -58,33 +45,21 @@ module g2grids
 
       contains
 
+!>    @brief This function returns the number of Grid Definition
```
Comment (edwardhartnett) [inline src/g2grids.f]: You don't need this doxygen block here.

We need doxygen blocks:
* Once for the file, at the top.
* Once for each module, above the module declaraion.
* Once for each function/subroutine.

File: src/g2grids.f [hunk 3]
```diff
@@ -58,33 +45,21 @@ module g2grids
...
+!>    Templates read in.
+!>    @author Gilbert         ORG: W/NP11    @date 2001-06-28
+!>
+
+!>    This function reads the list of GDT entries in the file
+!>    associated with fortran unit, lunit.  All the entries are stored in a
+!>    linked list called gridlist.
+!>
+!>    @param[in] lunit Fortran unit number associated the the GDT file.
+!>    @return - > The number of Grid Definition Templates read in.
```
Comment (edwardhartnett) [inline src/g2grids.f]: Take away the '- >' in the @return tag. We don't need to provide any extra text formatting stuff. So it should be:

` @return The number of Grid Definition Templates read in.
`

File: src/g2grids.f [hunk 4]
```diff
@@ -158,42 +133,30 @@ integer function readgrids(lunit)
 
          end function
 
+!>    @brief This subroutine searches a file referenced by an assigned number.
```
Comment (edwardhartnett) [inline src/g2grids.f]: This doxygen block is not needed.

File: src/g2grids.f [hunk 5]
```diff
@@ -158,42 +133,30 @@ integer function readgrids(lunit)
...
+!>    The input file format is described at the top of this module.
+!>
+!>    @param[in] lunit Unit number of file containing Grid definitions
+!>    @param[in] number Grid number of the requested Grid definition
+!>    @param[out] igdtn NN, indicating the number of the Grid Definition
+!>    Template 3.NN
+!>    @param[out] igdtmpl An array containing the values of each entry in
+!>    the Grid Definition Template.
+!>    @param[out] iret Error return code.
+!>    - 0 = no error
```
Comment (edwardhartnett) [inline src/g2grids.f]: Take out the '=' in the list. So:
`   - 0 no error
`

File: src/g2grids.f [hunk 6]
```diff
@@ -236,42 +199,30 @@ subroutine getgridbynum(lunit,number,igdtn,igdtmpl,iret)
  
          end subroutine
 
+!>    @brief This subroutine searches a file referenced by name.
+!>    @author Gilbert ORG: W/NP11 @date 2004-04-26
+!>
```
Comment (edwardhartnett) [inline src/g2grids.f]: THis doxygen block not needed.

File: src/g2grids.f [hunk 7]
```diff
@@ -236,42 +199,30 @@ subroutine getgridbynum(lunit,number,igdtn,igdtmpl,iret)
...
+!>    The input file format is described at the top of this module.
+!>
+!>    @param[in] lunit Unit number of file containing Grid definitions
+!>    @param[in] name Grid name of the requested Grid definition
+!>    @param[out] igdtn NN, indicating the number of the Grid Definition
+!>    Template 3.NN
+!>    @param[out] igdtmpl An array containing the values of each entry in
+!>    the Grid Definition Template.
+!>    @param[out] iret Error return code.
+!>    - 0 = no error
```
Comment (edwardhartnett) [inline src/g2grids.f]: Take out '=' signs in the list.

File: src/gb_info.f [hunk 8]
```diff
@@ -1,69 +1,57 @@
+!>    @file
+!>    @brief This subroutine searches the number of gridded fields and
+!>    the number of Local Use Sections through a GRIB2 message.
+!>    @author Gilbert ORG: W/NP11 @date 2000-05-25
+!>
```
Comment (edwardhartnett) [inline src/gb_info.f:5]: This doxygen block not needed.

File: src/gb_info.f [hunk 9]
```diff
@@ -1,69 +1,57 @@
...
+!>    returns the number of gridded fields found in the message and
+!>    the number (and maximum size) of Local Use Sections.
+!>    Also various checks  are performed to see if the message
+!>    is a valid GRIB2 message.
+!>
+!>    @param[in] cgrib Character array that contains the GRIB2 message.
+!>    @param[in] lcgrib Length (in bytes) of GRIB message in array cgrib.
+!>    @param[out] listsec0 Contains information decoded from GRIB Indicator Section 0.
+!>    Must be dimensioned >= 2.
+!>    - listsec0(1)=Discipline-GRIB Master Table Number (see Code Table 0.0)
```
Comment (edwardhartnett) [inline src/gb_info.f:17]: Take out = signs in lists.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: src/gb_info.f (hunk 9 above)
Comment (Hang-Lei-NOAA) [reply src/gb_info.f:17]: All these have been changed.

**Comment by GeorgeGayno-NOAA** [comment]
@Hang-Lei-NOAA I am getting compiler warnings about g2grid.f:

_warning: Member maxtemp (variable) of class g2grids is not documented.
warning: Member num_grids (variable) of class g2grids is not documented._

It is an easy fix. Let me know if you need assistance.

**Comment by Hang-Lei-NOAA** [comment]
Hi, George
I add @param for both Does this works?
Thanks

**Comment by Hang-Lei-NOAA** [comment]
Hi, George

Let me know how to deal with this warning.
I added the @param,  which is not working.

Thanks,
Hang

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**